src/models/ProfilePicture.ts

```diff
@@ -52,6 +54,32 @@ export async function generateImage(address: string) {
     const readableStream = await generateAndDownloadImage(nickname, 3)
     const { cid } = await uploadToIpfs(readableStream)
     picture.cid = cid
+    picture.newCid = cid
+    picture.isFinished = true
+    await picture.save()
+  } catch (e) {
+    console.error(e)
+  } finally {
+    picture.generating = false
+    await picture.save()
+  }
+
+  return picture
+}
+
+export async function generateNewImage(address: string) {
+  const picture = await ProfilePictureModel.findOne({ address })
+
+  if (!picture) return null
+  if (picture.generating) return picture
```

Comment:
🤔 
Should we handle it on the frontend somehow?

Reply:
Nope